internal/controller/base_controller.go

```diff
@@ -163,6 +164,12 @@ func CtrlResultForErr(err error) (ctrl.Result, error) {
 			return ctrl.Result{RequeueAfter: time.Minute}, nil
 		case nerr.StatusCode == http.StatusNotFound:
 			return ctrl.Result{}, err
+		case ngrok.IsErrorCode(nerr, ngrokapi.NgrokOpErrFailedToCreateCSR.Code):
+			return ctrl.Result{RequeueAfter: 30 * time.Second}, nerr
+		case ngrok.IsErrorCode(nerr, ngrokapi.NgrokOpErrFailedToCreateUpstreamService.Code, ngrokapi.NgrokOpErrFailedToCreateTargetService.Code):
+			return ctrl.Result{RequeueAfter: 1 * time.Minute}, nerr
+		case ngrok.IsErrorCode(nerr, ngrokapi.NgrokOpErrEndpointDenied.Code):
```

Comment:
this won't be relevant anymore, but we can remove it after we remove the enriched error on the other side